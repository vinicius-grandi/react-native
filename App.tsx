// import { StatusBar } from 'expo-status-bar';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';
import Main from './src/components/Main';
import { styles as imStyles } from './src/components/Form';

export const styles = StyleSheet.create({
  container: {
    fontSize: '50px',
    flex: 0.9,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    width: '80%',
    alignSelf: 'center',
    height: '100%',
  },
  main: {
    flex: 1,
    backgroundColor: '#e6e6e6',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default function App() {
  return (
    <View style={styles.main}>
      <Text style={imStyles.title}>onebithealth</Text>
      <View style={styles.container}>
        <Main />
      </View>
    </View>
  );
}
