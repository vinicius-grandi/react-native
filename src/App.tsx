// import { StatusBar } from 'expo-status-bar';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { registerRootComponent } from 'expo';
import Main from './components/Main';
import { styles as imStyles } from './components/Form';

export const styles = StyleSheet.create({
  container: {
    fontSize: '50px',
    flex: 0.9,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    width: '80%',
    alignSelf: 'center',
    height: '100%',
  },
  main: {
    flex: 1,
    backgroundColor: '#e6e6e6',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default function App() {
  return (
    <View style={styles.main}>
      <Text style={imStyles.title}>onebithealth</Text>
      <View style={styles.container}>
        <Main />
      </View>
    </View>
  );
}

registerRootComponent(App);
