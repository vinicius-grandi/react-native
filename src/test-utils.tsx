import { render, fireEvent, screen } from '@testing-library/react-native';

const customRender: typeof render = (ui, options) => render(ui, options);

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render, fireEvent, screen };
