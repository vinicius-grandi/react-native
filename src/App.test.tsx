import App from './components/Main';
import { render, screen, fireEvent } from './test-utils';

describe('App', () => {
  it('render two text inputs where you can type and calculate your body mass index', () => {
    const s = render(<App />).toJSON();
    const heightInput = screen.getByPlaceholderText('ex: 1.72');
    const weightInput = screen.getByPlaceholderText('ex: 60');

    fireEvent.changeText(heightInput, '1.72');
    fireEvent.changeText(weightInput, '60');

    const calculateBtn = screen.getByRole('button');
    fireEvent.press(calculateBtn);
    expect(s).toMatchSnapshot();
  });
});
