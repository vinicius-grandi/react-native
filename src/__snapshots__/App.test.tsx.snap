// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render two text inputs where you can type and calculate your body mass index 1`] = `
<View
  style={
    Object {
      "borderRadius": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text>
      Altura
    </Text>
    <TextInput
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="ex: 1.72"
      style={
        Object {
          "backgroundColor": "#e6e6e6",
          "color": "#000",
          "textAlign": "center",
          "width": "90%",
        }
      }
    />
    <Text>
      Peso
    </Text>
    <TextInput
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="ex: 60"
      style={
        Object {
          "backgroundColor": "#e6e6e6",
          "color": "#000",
          "textAlign": "center",
          "width": "90%",
        }
      }
    />
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#e21536",
                },
              ]
            }
          >
            Calcular
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
